File: Java/Q41.java
import java.util.Scanner;

public class Q41 {
    public static String reverse(String sentence) {
        if (sentence.length() <= 1) {
            return sentence;
        }

        return reverse(sentence.substring(1)) + sentence.charAt(0);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a sentence: ");
        String sentence = sc.nextLine();

        String reversed = reverse(sentence);
        System.out.println("Reversed sentence: " + reversed);

        sc.close();
    }
}

File: Python/Q41.py
# Q41.py - Program to Reverse a Sentence Using Recursion
def reverse_sentence(s):
    if len(s) == 0:
        return s
    else:
        return reverse_sentence(s[1:]) + s[0]

sentence = input("Enter a sentence: ")
print(f"Reversed: {reverse_sentence(sentence)}")